### PR TITLE
Fix the filtering of stream in descending order by timestamp

### PR DESCRIPTION
**Pull request by ButterBright**
When filtering according to the indexRule while sorting in descending order by timestamp,  the elements remaining after applying the limit are incorrect.

- [x] Explain briefly why the bug exists and how to fix it.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-banyandb/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you describe what is the effect of this bug?

**Comment by ButterBright** [comment]
Sure.

**Review by hanahmily — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Release Notes.
 - Fix duplicated items in the query aggregation top-n list.
 - Fix non-"value" field in topN pre-calculation result measure is lack of data.
 - Encode escaped characters to int64 bytes to fix the malformed data.
+- Fix the filtering of stream in descending order by timestamp.
```
Comment (hanahmily) [inline CHANGES.md:22]: Move it to 0.7.0

**Review by hanahmily — CHANGES_REQUESTED** [review]

**Comment by ButterBright** [comment]
When sorting in descending order by timestamp, the previous method always filtered out the elements with the smallest timestamps, rather than the largest ones.

**Review by hanahmily — APPROVED** [review]

**Pull request merged**